ray-operator/controllers/common/pod.go

```diff
@@ -113,6 +132,57 @@ func BuildPod(podTemplateSpec v1.PodTemplateSpec, rayNodeType rayiov1alpha1.RayN
 	return pod
 }
 
+// BuildAutoscalerContainer build a ray autoscaler container which can be appended to head pod.
+func BuildAutoscalerContainer(redisPasswd string) v1.Container {
+	container := v1.Container{
+		Name: "autoscaler",
+		// TODO: choose right version based on instance.spec.Version
+		Image:           "kuberay/autoscaler:nightly",
+		ImagePullPolicy: v1.PullAlways,
+		Env: []v1.EnvVar{
+			{
+				Name: "RAY_CLUSTER_NAME",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						FieldPath: "metadata.labels['ray.io/cluster']",
+					},
+				},
+			},
+			{
+				Name: "RAY_CLUSTER_NAMESPACE",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						FieldPath: "metadata.namespace",
+					},
+				},
+			},
+		},
+		Command: []string{
+			"/home/ray/anaconda3/bin/python",
+		},
+		Args: []string{
+			"/home/ray/run_autoscaler_with_retries.py",
+			"--cluster-name",
+			"$(RAY_CLUSTER_NAME)",
+			"--cluster-namespace",
+			"$(RAY_CLUSTER_NAMESPACE)",
+			"--redis-password",
+			redisPasswd,
+		},
+		Resources: v1.ResourceRequirements{
+			Limits: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("500m"),
```

Comment:
We _might_ eventually want to make the requests and limits configurable.

Reply:
Agree. This should be configurable eventually.

Reply:
I added a TODO here and we can address it in later release